### DISCUSSION TITLE
How can we load test websocket?

### DISCUSSION THREAD

**Discussion by neerajdlh74** [miguelgrinberg/python-socketio#932] [Q&A] [answered]
i want to load test my websocket to test how many connections it can support simultaneously.

**Comment by miguelgrinberg** [comment, marked as answer]
You can find a load testing application or service that supports Socket.IO (there aren't that many, sadly) or else write your own load testing client.

**Reply by neerajdlh74** [reply]
Thanks for replying, can i use locust for load test client?

**Reply by miguelgrinberg** [reply]
I don't have experience with Locust, so I do not know if it is compatible. As long as they support a version of the Socket.IO protocol that is compatible with the one used by your application everything should work.

**Reply by neerajdlh74** [reply]
Okay, can you suggest application from your side that you have used for load testing and compatible with python-socketio?

**Reply by miguelgrinberg** [reply]
Sorry, I don't have anything to recommend. I write my own load testing code, but I haven't done this for any of my open source apps.

**Reply by neerajdlh74** [reply]
Okay, thank you for replying.

**Reply by neerajdlh74** [reply]
Hi, I need the websocket to handle 5000 clients while emitting event for every 2 seconds to all users. how i can achieve it through python-socketio?

**Reply by neerajdlh74** [reply]
right now i am not able to connect even 1000 clients with messages. the load on cpu is upto 100% and clients are disconnected.

**Reply by miguelgrinberg** [reply]
python-socketio does not have any limits on number of concurrent users. If you max out with one CPU and one server process at 1000 users, then 5 CPUs with a process on each should allow you do go up to 5000 more or less.
